pairtools/pairtools_stats.py

```diff
@@ -1,113 +1,115 @@
-        ' all overlapping statistics. Non-overlapping statistics are appended to'
-        ' the end of the file.',
-    )
-
+    help="If specified, merge multiple input stats files instead of calculating"
+    " statistics of a .pairs/.pairsam file. Merging is performed via summation of"
+    " all overlapping statistics. Non-overlapping statistics are appended to"
+    " the end of the file.",
+)
 @common_io_options
-
 def stats(input_path, output, merge, **kwargs):
-    '''Calculate pairs statistics. 
+    """Calculate pairs statistics. 
 
     INPUT_PATH : by default, a .pairs/.pairsam file to calculate statistics.
     If not provided, the input is read from stdin.
     If --merge is specified, then INPUT_PATH is interpreted as an arbitrary number 
     of stats files to merge.
     
     The files with paths ending with .gz/.lz4 are decompressed by bgzip/lz4c. 
-    '''
+    """
     stats_py(input_path, output, merge, **kwargs)
 
+
 def stats_py(input_path, output, merge, **kwargs):
     if merge:
         do_merge(output, input_path, **kwargs)
         return
 
-    instream = (_fileio.auto_open(input_path[0], mode='r', 
-                                  nproc=kwargs.get('nproc_in'),
-                                  command=kwargs.get('cmd_in', None)) 
-                if input_path else sys.stdin)
-    outstream = (_fileio.auto_open(output, mode='w', 
-                                   nproc=kwargs.get('nproc_out'),
-                                   command=kwargs.get('cmd_out', None)) 
-                 if output else sys.stdout)
-
+    instream = (
+        _fileio.auto_open(
+            input_path[0],
+            mode="r",
+            nproc=kwargs.get("nproc_in"),
+            command=kwargs.get("cmd_in", None),
+        )
+        if input_path
+        else sys.stdin
+    )
+    outstream = (
+        _fileio.auto_open(
+            output,
+            mode="w",
+            nproc=kwargs.get("nproc_out"),
+            command=kwargs.get("cmd_out", None),
+        )
+        if output
+        else sys.stdout
+    )
 
     header, body_stream = _headerops.get_header(instream)
-
+    cols = _headerops.extract_column_names(header)
 
     # new stats class stuff would come here ...
     stats = PairCounter()
 
     # Collecting statistics
-    for line in body_stream:
-        cols = line.rstrip().split(_pairsam_format.PAIRSAM_SEP)
 
-        # algn1:
-        chrom1 = cols[_pairsam_format.COL_C1]
-        pos1 = int(cols[_pairsam_format.COL_P1])
-        strand1 = cols[_pairsam_format.COL_S1]
-        # algn2:
-        chrom2 = cols[_pairsam_format.COL_C2]
-        pos2 = int(cols[_pairsam_format.COL_P2])
-        strand2 = cols[_pairsam_format.COL_S2]
-        # pair type:
-        pair_type = cols[_pairsam_format.COL_PTYPE]
+    for chunk in pd.read_table(body_stream, names=cols, chunksize=100_000):
+        stats.add_pairs_from_dataframe(chunk)
+    # for line in body_stream:
```

Comment:
This looks good, can we now remove an old code?

Reply:
I think so! Kept it for now just in case...